### PR TITLE
[PORT-8328] [GCP] Changed fetching to be of latest resource version

### PR DESCRIPTION
**Pull request by matan84**
# Description

What - Changed resource fetching to be only on last resource version of the asset
Why - Parsing the asset for each property is redundant
How - Getting the response of the asset, from it return the latest resource

## Type of change

Please leave one option from the following and delete the rest:

- [ ] New feature (non-breaking change which adds functionality)

## API Documentation

https://docs.getport.io/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/

### CODE REVIEW

**Review by yairsimantov20 — CHANGES_REQUESTED** [review]
🌊

File: integrations/gcp/gcp_core/search/resource_searches.py [hunk 1]
```diff
@@ -63,9 +58,15 @@ async def search_all_resources_in_project(
...
-                    logger.info(f"Found {len(resources)} {asset_type}'s")
+                raw_assets = parse_protobuf_messages(paginated_response.results)
+                assets = typing.cast(list[AssetData], raw_assets)
+                if assets:
+                    logger.info(f"Found {len(assets)} {asset_type}'s")
+                    resources = []
+                    for asset in assets:
+                        resource = parse_latest_resource_from_asset(asset)
+                        resource[EXTRA_PROJECT_FIELD] = project_name
+                        resources.append(resource)
```
Comment (Tankilevitch) [inline integrations/gcp/gcp_core/search/resource_searches.py:69]:
```suggestion
                        latest_resource = parse_latest_resource_from_asset(asset)
                        latest_resource[EXTRA_PROJECT_FIELD] = project_name
                        latest_resources.append(resource)
```

File: integrations/gcp/gcp_core/search/resource_searches.py [hunk 2]
```diff
@@ -98,7 +99,7 @@ async def list_all_topics_per_project(
                         f"Found {len(topics)} {AssetTypesWithSpecialHandling.TOPIC}'s"
                     )
                     for topic in topics:
-                        topic[EXTRA_PROJECT_FIELD] = project
+                        topic[EXTRA_PROJECT_FIELD] = project_name
```
Comment (Tankilevitch) [inline integrations/gcp/gcp_core/search/resource_searches.py:102]: can we provide it with all the project information? and not only the name?

File: integrations/gcp/main.py [hunk 3]
```diff
@@ -130,7 +131,17 @@ async def feed_events_callback(request: Request) -> Response:
...
+            f"Didn't find any {asset_type} resource named: {asset_name}. Deleting ocean entity."
+        )
+        await ocean.unregister(
+            [
+                Entity(
+                    blueprint=asset_type,
+                    identifier=asset_name,
+                )
+            ]
+        )
```
Comment (Tankilevitch) [inline integrations/gcp/main.py:144]: I think you don't need to do this with @omby8888 feature of delete on event, please verify

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: integrations/gcp/.port/resources/blueprints.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
     }
   },
   {
-    "identifier": "functions",
+    "identifier": "gcpCloudFunctions",
```
Comment (Tankilevitch) [inline integrations/gcp/.port/resources/blueprints.json:51]: remove

File: integrations/gcp/.port/resources/blueprints.json [hunk 5]
```diff
@@ -11,7 +11,7 @@
     "aggregationProperties": {}
   },
   {
-    "identifier": "containerCluster",
+    "identifier": "gcpContainerCluster",
```
Comment (Tankilevitch) [inline integrations/gcp/.port/resources/blueprints.json:14]: remove

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**